lib/paths.coffee

```diff
@@ -1,139 +1,31 @@
-    "/usr/local/share/jupyter",
-    "/usr/share/jupyter"
-  ]
-
-ENV_JUPYTER_PATH = [path.join(sysPrefix, 'share', 'jupyter')]
-
-# Returns the home specified by environment variable or
-# node's built in path.resolve('~')
-userHome = ->
-  homeDir = process.env['HOME'] or process.env['USERPROFILE'] or path.resolve('~')
-  homeDir = fs.realpathSync(homeDir)
-  return homeDir
-
-# Get the Jupyter config directory for this platform and user.
-# Reutrns env[JUPYTER_CONFIG_DIR] if defined, else ~/.jupyter
-jupyterConfigDir = ->
-  homeDir = userHome()
-  if process.env['JUPYTER_CONFIG_DIR']
-    return process.env['JUPYTER_CONFIG_DIR']
-  return path.join(homeDir, '.jupyter')
-
-# Get the config directory for Jupyter data files.
-#
-# These are non-transient, non-configuration files.
-#
-# Returns process.env[JUPYTER_DATA_DIR] if defined,
-# else a platform-appropriate path.
-jupyterDataDir = ->
-  if process.env['JUPYTER_DATA_DIR']
-    return process.env['JUPYTER_DATA_DIR']
-  else if process.platform == 'win32'
-    appData = process.env['APPDATA']
-    if appData
-      return path.join appData, 'jupyter'
-    else
-      return path.join jupyterConfigDir(), 'data'
-  else if process.platform == 'darwin'
-    return path.join(userHome(), 'Library', 'Jupyter')
-  else
-    # Linux, non-OS X Unix, AIX, etc.
-    xdg = process.env['XDG_DATA_HOME']
-    if not xdg
-      xdg = path.join(userHome(), '.local', 'share')
-    return path.join(xdg, 'jupyter')
 
-
-# Returns the path for ~/.ipython
 ipythonDataDir = ->
-  return path.join(userHome(), '.ipython')
-
-# Return the runtime dir for transient jupyter files.
-#
-# Returns process.env[JUPYTER_RUNTIME_DIR] if defined.
-#
-# Respects XDG_RUNTIME_DIR on non-OS X, non-Windows,
-#   falls back on data_dir/runtime otherwise.
-jupyterRuntimeDir = ->
-  if process.env['JUPYTER_RUNTIME_DIR']
-    return process.env['JUPYTER_RUNTIME_DIR']
-
-  if process.platform == 'darwin' or process.platform == 'win32'
-    return path.join(jupyterDataDir(), 'runtime')
-  else
-    # Linux, non-OS X Unix, AIX, etc.
-    xdg = process.env['XDG_RUNTIME_DIR']
-    if xdg
-      # Ref https://github.com/jupyter/jupyter_core/commit/176b7a9067bfc20bfa97be5dae93912e2930a427#commitcomment-11331375
-      return path.join(xdg, 'jupyter')
-    return path.join(jupyterDataDir, 'runtime')
+    child_process.spawnSync('ipython',['locate']).stdout.toString().replace '\n',''
 
-# Return the list of directories to search
-#
-# JUPYTER_PATH environment variable has highest priority.
-#
-# If subdirs are given, that subdirectory path will be added to each element.
-# Examples:
-#
-# > jupyterPath()
-# ['/Users/rgbkrk/.local/jupyter', '/usr/local/share/jupyter']
-#
-# >jupyterPath
-# ['/Users/rgbkrk/.local/jupyter/kernels', '/usr/local/share/jupyter/kernels']
-jupyterPath = (subdirs...) ->
-  paths = []
+jupyterResolve = _.memoize (category) ->
+    match = child_process.spawnSync('jupyter', ['--paths']).stdout.toString()
```

Comment:
There's now a `--json` flag which was created because of this use case in Atom, node, and Electron. I'd like to see this get addressed in https://github.com/nteract/jupyter-paths and just rely on it there (more than one package needs this functionality).